webview-ui/src/components/common/MermaidBlock.tsx

```diff
@@ -212,6 +294,39 @@ const LoadingMessage = styled.div`
 	font-size: 0.9em;
 `
 
+const CodeBlock = styled.pre`
+	background-color: var(--vscode-editor-background);
+	border: 1px solid var(--vscode-editor-lineHighlightBorder);
+	border-radius: 3px;
+	padding: 8px;
+	overflow: auto;
+	max-height: 200px;
+
+	code {
+		font-family: var(--vscode-editor-font-family);
+		font-size: var(--vscode-editor-font-size);
+		white-space: pre-wrap;
+		word-break: break-all;
+	}
+`
+
+const CopyButton = styled.button`
```

Comment:
Is there not an existing component for this?

Reply:
Not in a shared place, I don't think. Probably should be though.